src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -101,6 +104,31 @@ private void CheckFactoryWasUnregisteredFromDataSource(SnapshotFactory factory,
             mockSonarErrorDataSource.Verify(x => x.RemoveFactory(factory), times);
         }
 
+        [TestMethod]
+        public void OnBufferChange_ErrorListIsUpdated()
+        {
+            // Use the test version of the text buffer to bypass the span translation code
+            testSubject = new TestableTextBufferIssueTracker(taggerProvider.dte, taggerProvider,
+                mockedJavascriptDocumentFooJs.Object, javascriptLanguage, issuesFilter.Object,
+                mockSonarErrorDataSource.Object, new TestLogger());
+
+
+            var beforeSnapshot = CreateMockTextSnapshot(100, "foo", 1);
+            var afterSnapshot = CreateMockTextSnapshot(100, "bar", 1);
+
+            // Need to register a tagger for the buffer to start listening to buffer change events
+            using (testSubject.CreateTagger())
+            {
+                mockAnalyzerController.Invocations.Clear();
+                mockSonarErrorDataSource.Invocations.Clear();
+
+                RaiseBufferChangedEvent(mockDocumentTextBuffer, beforeSnapshot.Object, afterSnapshot.Object);
+
+                CheckAnalysisWasNotRequested();
+                CheckErrorListRefreshWasRequested(1);
```

Comment:
This is unrelated and was here before, but maybe it's worth changing anyway if you're doing other changes: CheckErrorListRefreshWasRequested is always accepting 1 as an argument, so it's redundant. I had to go and look what it's doing, as I thought it was number of expected issues or something.